app/src/main/kotlin/games/BoardElement.kt

```diff
@@ -1,6 +1,6 @@
 package games
 
-class BoardElement  {    // put row and column in the constructor
+data class BoardElement(val row: Int, val col: Int) {
```

Comment:
This is an important change : the board element knows where it is on the board rather than having its position implied by the row/col in the array.  This is going to make things much better as we implement AI.